src/layout/boxmodel.ts

```diff
@@ -71,22 +71,24 @@ function update(id: number, box: number[]): void {
 
 export function layout(element: Element, x: number = null, y: number = null): number[] {
     let box: number[] = null;
-    let rect = element.getBoundingClientRect();
-    x = x !== null ? x : ("pageXOffset" in window ? window.pageXOffset : document.documentElement.scrollLeft);
-    y = y !== null ? y : ("pageYOffset" in window ? window.pageYOffset : document.documentElement.scrollTop);
+    if (typeof element.getBoundingClientRect === "function") {
```

Comment:
>getBoundingClientRect [](start = 23, length = 21)

is any of the boxmodel code worth running without this? wondering if initial wireup should check for getBoundingClientRect and not enable boxmodel if it doesnt exist

Reply:
+1. Also can we send a signal that this situation happened so we can tell that the data is missing "by design" because the browser doesn't support it?

---
In reply to: [347530487](https://github.com/microsoft/clarity-js/pull/160#discussion_r347530487) [](ancestors = 347530487)

Reply:
This is a fair point. What I have right now is more of a mitigation. One known case is when we have a text node instead of an element, and not so much to do with browser. I will tackle this in subsequent ones, together with selection event. We probably will have to walk up the DOM chain to get an element as target and not text nodes - which are hard to track in our current model (e.g. there's no selector).